### ISSUE TITLE
Crash when marking a test with custom mark called integration

### ISSUE THREAD

**Issue by jramosss** [pytest-dev/pytest#10164]
- [X] a detailed description of the bug or problem you are having
I ran into a problem when trying to mark a class with a custom mark.
I declared it in my pytest.ini as
```py
markers =
    integration: Run integration tests
```
And marked my class as:
```py
@pytest.mark.integration
class TestStripeIntegration:
```

And when i run: 
`pytest -s --collect-only -v -m "not integration"`
I get all my tests (included the ones with the integration mark) and this error message:
```
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/_pytest/main.py", line 268, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/_pytest/main.py", line 321, in _main
INTERNALERROR>     config.hook.pytest_collection(session=session)
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
INTERNALERROR>     self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/_pytest/main.py", line 332, in pytest_collection
INTERNALERROR>     session.perform_collect()
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/_pytest/main.py", line 660, in perform_collect
INTERNALERROR>     hook.pytest_collection_modifyitems(
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
INTERNALERROR>     self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/_pytest/mark/__init__.py", line 249, in pytest_collection_modifyitems
INTERNALERROR>     deselect_by_mark(items, config)
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/_pytest/mark/__init__.py", line 236, in deselect_by_mark
INTERNALERROR>     config.hook.pytest_deselected(items=deselected)
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
INTERNALERROR>     self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/_pytest/terminal.py", line 494, in pytest_deselected
INTERNALERROR>     self._add_stats("deselected", items)
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/_pytest/terminal.py", line 463, in _add_stats
INTERNALERROR>     self._set_main_color()
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/_pytest/terminal.py", line 1168, in _set_main_color
INTERNALERROR>     self._main_color = self._determine_main_color(bool(unknown_types))
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/_pytest/terminal.py", line 1155, in _determine_main_color
INTERNALERROR>     elif "passed" in stats or not self._is_last_item:
INTERNALERROR>   File "/home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/_pytest/terminal.py", line 578, in _is_last_item
INTERNALERROR>     assert self._session is not None
INTERNALERROR> AssertionError
```
And i cannot run my tests because of this error.

- [X] output of `pip list` from the virtual environment you are using
```
Package                           Version    Location                                                                              
--------------------------------- ---------- --------------------------------------------------------------------------------------
aiohttp                           3.8.1      
aiosignal                         1.2.0      
appdirs                           1.4.4      
argcomplete                       1.12.3     
argon2-cffi                       21.3.0     
argon2-cffi-bindings              21.2.0     
asgiref                           3.4.1      
async-timeout                     4.0.2      
atomicwrites                      1.4.0      
attrs                             21.2.0     
autopep8                          1.6.0      
backcall                          0.2.0      
backports.entry-points-selectable 1.1.1      
better-admin                      1.0        /home/shuls/Documents/seenka/repos/workspaces/.venv/src/better-admin/libs/better_admin
bleach                            4.1.0      
cachetools                        4.2.4      
certifi                           2021.10.8  
cffi                              1.15.0     
chardet                           3.0.4      
charset-normalizer                2.0.9      
click                             8.0.3      
colorlog                          4.8.0      
coverage                          4.5.4      
cryptography                      36.0.1     
curlify                           2.2.1      
debugpy                           1.5.1      
decorator                         5.1.0      
defusedxml                        0.7.1      
dictdiffer                        0.9.0      
diff-match-patch                  20200713   
distlib                           0.3.4      
Django                            3.1.7      
django-admin-list-filter-dropdown 1.0.3      
django-admin-rangefilter          0.8.3      
django-admin-shortcuts            2.0.0      
django-autocompletefilter         0.0.8      
django-cas-ng                     4.2.1      
django-cors-headers               3.0.2      
django-coverage-plugin            2.0.2      
django-elasticsearch-dsl          7.1.4      
django-environ                    0.4.5      
django-extensions                 2.2.5      
django-filter                     2.4.0      
django-import-export              2.7.0      
django-logentry-admin             1.0.5      /home/shuls/Documents/seenka/repos/workspaces/.venv/src/django-logentry-admin         
django-nose                       1.4.7      
django-prometheus                 2.2.0      
django-queryset-csv               1.1.0      
django-redis                      4.10.0     
django-smtp-ssl                   1.0        
django-storages                   1.7.2      
django-test-without-migrations    0.6        
django-timezone-field             4.2.1      
django-username-email             2.4.2      
djangorestframework               3.12.4     
djangorestframework-api-key       2.0.0      
djangorestframework-simplejwt     4.3.0      
docopt                            0.6.2      
elasticsearch                     7.1.0      
elasticsearch-dsl                 7.1.0      
entrypoints                       0.3        
et-xmlfile                        1.1.0      
execnet                           1.9.0      
facebook-business                 9.0.0      
Faker                             10.0.0     
filelock                          3.4.0      
frozenlist                        1.2.0      
google-ads                        11.0.2     
google-api-core                   1.23.0     
google-api-python-client          1.12.8     
google-auth                       1.23.0     
google-auth-httplib2              0.0.4      
google-auth-oauthlib              0.4.2      
google-cloud-bigquery             3.2.0      
google-cloud-bigquery-storage     2.13.2     
google-cloud-core                 2.2.1      
google-cloud-pubsub               2.2.0      
google-cloud-storage              1.43.0     
google-crc32c                     1.3.0      
google-resumable-media            2.1.0      
googleapis-common-protos          1.52.0     
grpc-google-iam-v1                0.12.3     
grpcio                            1.43.0     
grpcio-status                     1.46.3     
gunicorn                          19.9.0     
httplib2                          0.20.2     
idna                              2.8        
importlib-resources               5.4.0      
inflect                           5.3.0      
iniconfig                         1.1.1      
ipdb                              0.12.2     
ipykernel                         6.6.0      
ipython                           7.30.1     
ipython-genutils                  0.2.0      
jedi                              0.18.1     
Jinja2                            3.0.3      
jsonschema                        4.3.2      
jupyter-client                    7.1.0      
jupyter-core                      4.9.1      
jupyterlab-pygments               0.1.2      
libcst                            0.3.23     
libretranslatepy                  2.1.1      
lxml                              4.7.1      
MarkupPy                          1.14       
MarkupSafe                        2.0.1      
matplotlib-inline                 0.1.3      
mistune                           0.8.4      
more-itertools                    8.12.0     
multidict                         5.2.0      
mypy                              0.740      
mypy-extensions                   0.4.3      
nbclient                          0.5.9      
nbconvert                         6.3.0      
nbformat                          5.1.3      
nest-asyncio                      1.5.4      
nose                              1.3.7      
notebook                          6.4.6      
nox                               2020.12.31 
numpy                             1.21.5     
oauth2client                      4.1.3      
oauthlib                          3.1.1      
odfpy                             1.4.1      
openpyxl                          3.0.9      
packaging                         21.3       
pandas                            1.3.5      
pandocfilters                     1.5.0      
parso                             0.8.3      
pathspec                          0.9.0      
pep562                            1.1        
pexpect                           4.8.0      
phonenumbers                      8.12.44    
pickleshare                       0.7.5      
Pillow                            8.4.0      
pip                               20.0.2     
pkg-resources                     0.0.0      
places                            1.1        /home/shuls/Documents/seenka/repos/workspaces/.venv/src/places/libs/places            
platformdirs                      2.4.0      
pluggy                            0.13.1     
progress                          1.6        
prometheus-client                 0.12.0     
prompt-toolkit                    3.0.24     
proto-plus                        1.19.8     
protobuf                          3.20.1     
psycopg2-binary                   2.9.2      
ptyprocess                        0.7.0      
py                                1.11.0     
pyarrow                           8.0.0      
pyasn1                            0.4.8      
pyasn1-modules                    0.2.8      
pycodestyle                       2.8.0      
pycountry                         20.7.3     
pycparser                         2.21       
Pygments                          2.10.0     
PyJWT                             2.3.0      
pykwalify                         1.7.0      
pyparsing                         3.0.6      
pypelines                         0.1        /home/shuls/Documents/seenka/repos/workspaces/.venv/src/pypelines                     
pyrsistent                        0.18.0     
pytest                            7.1.2      
pytest-django                     3.6.0      
pytest-forked                     1.4.0      
pytest-lazy-fixture               0.6.3      
pytest-mock                       3.8.1      
pytest-sugar                      0.9.2      
pytest-xdist                      2.5.0      
python-cas                        1.6.0      
python-dateutil                   2.8.2      
python-slugify                    4.0.0      
pytz                              2019.3     
PyYAML                            5.4.1      
pyzmq                             22.3.0     
redis                             3.3.11     
requests                          2.22.0     
requests-oauthlib                 1.3.0      
rsa                               4.8        
Send2Trash                        1.8.0      
sentry-sdk                        1.5.1      
setuptools                        44.0.0     
six                               1.16.0     
sqlparse                          0.4.2      
stripe                            3.5.0      
tablib                            3.1.0      
tags                              1.0        /home/shuls/Documents/seenka/repos/workspaces/.venv/src/tags/libs/tags                
tblib                             1.7.0      
termcolor                         1.1.0      
terminado                         0.12.1     
testpath                          0.5.0      
text-unidecode                    1.3        
toml                              0.10.2     
tomli                             2.0.1      
tornado                           6.1        
traitlets                         5.1.1      
translate                         3.6.1      
twilio                            7.6.0      
typed-ast                         1.4.3      
typing-extensions                 4.0.1      
typing-inspect                    0.7.1      
unicodecsv                        0.14.1     
uritemplate                       3.0.1      
urllib3                           1.25.11    
users                             1.0        /home/shuls/Documents/seenka/repos/workspaces/.venv/src/users/libs/users              
uuid                              1.30       
validators                        0.19.0     
virtualenv                        20.10.0    
wcwidth                           0.2.5      
webencodings                      0.5.1      
Werkzeug                          2.0.2      
xlrd                              2.0.1      
xlwt                              1.3.0      
yarl                              1.7.2      
zipp                              3.6.0      
```
- [X] pytest and operating system versions
   - pytest 7.1.2
   - Ubuntu 20.04 LTS

**Comment by nicoddemus** [comment]
Hi @jramosss,

I cannot reproduce the problem, I even tried to install `pytest-sugar`, to see if it is related, but it still works.

`_session` really cannot be `None` at that point, because it is set during `pytest_sessionstart`, which is called much earlier than `pytest_collection`.

Can you share a minimal reproducible example?

**Comment by jramosss** [comment]
Hi @nicoddemus 
Well, i really tried to make a minimal example, but i really couldn't, its only within my (really large) project that crashes.
But, in a long shot try to not get this issue closed and left without answer, i saw that pytest_sessionstart function is not being called, i added an ipdb.set_trace() to try and trace the problem, dont know if that helps
```
(.venv) ➜  workspaces git:(workspace-cancel) ✗ pytest -s --collect-only -v -m "not integration"

> /home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/_pytest/main.py(268)wrap_session()
    267             set_trace()
--> 268             config.hook.pytest_sessionstart(session=session)
    269             initstate = 2

ipdb> s
--Call--
> /home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/_pytest/config/compat.py(36)__getattr__()
     35 
---> 36     def __getattr__(self, key, _wraps=functools.wraps):
     37         hook = getattr(self.__hook_caller, key)

ipdb> n
> /home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/_pytest/config/compat.py(37)__getattr__()
     36     def __getattr__(self, key, _wraps=functools.wraps):
---> 37         hook = getattr(self.__hook_caller, key)
     38         if key not in imply_paths_hooks:

ipdb> 
> /home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/_pytest/config/compat.py(38)__getattr__()
     37         hook = getattr(self.__hook_caller, key)
---> 38         if key not in imply_paths_hooks:
     39             self.__dict__[key] = hook

ipdb> 
> /home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/_pytest/config/compat.py(39)__getattr__()
     38         if key not in imply_paths_hooks:
---> 39             self.__dict__[key] = hook
     40             return hook

ipdb> 
> /home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/_pytest/config/compat.py(40)__getattr__()
     39             self.__dict__[key] = hook
---> 40             return hook
     41         else:

ipdb> 
--Return--
<_HookCaller ...sessionstart'>
> /home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/_pytest/config/compat.py(40)__getattr__()
     39             self.__dict__[key] = hook
---> 40             return hook
     41         else:

ipdb> 
--Call--
> /home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/hooks.py(271)__call__()
    270 
--> 271     def __call__(self, *args, **kwargs):
    272         if args:

ipdb> 
> /home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/hooks.py(272)__call__()
    271     def __call__(self, *args, **kwargs):
--> 272         if args:
    273             raise TypeError("hook calling supports only keyword arguments")

ipdb> 
> /home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/hooks.py(274)__call__()
    273             raise TypeError("hook calling supports only keyword arguments")
--> 274         assert not self.is_historic()
    275         if self.spec and self.spec.argnames:

ipdb> 
> /home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/hooks.py(275)__call__()
    274         assert not self.is_historic()
--> 275         if self.spec and self.spec.argnames:
    276             notincall = (

ipdb> n
> /home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/hooks.py(277)__call__()
    276             notincall = (
--> 277                 set(self.spec.argnames) - set(["__multicall__"]) - set(kwargs.keys())
    278             )

ipdb> 
> /home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/hooks.py(276)__call__()
    275         if self.spec and self.spec.argnames:
--> 276             notincall = (
    277                 set(self.spec.argnames) - set(["__multicall__"]) - set(kwargs.keys())

ipdb> 
> /home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/hooks.py(279)__call__()
    278             )
--> 279             if notincall:
    280                 warnings.warn(

ipdb> 
> /home/shuls/Documents/seenka/repos/workspaces/.venv/lib/python3.8/site-packages/pluggy/hooks.py(285)__call__()
    284                 )
--> 285         return self._hookexec(self, self.get_hookimpls(), kwargs)
    286 

ipdb> 
Test session starts (platform: linux, Python 3.8.10, pytest 7.1.2, pytest-sugar 0.9.2)
Django settings: service.settings.local (from environment variable)
rootdir: /home/shuls/Documents/seenka/repos/workspaces, configfile: pytest.ini
plugins: Faker-10.0.0, mock-3.8.1, sugar-0.9.2, forked-1.4.0, xdist-2.5.0, lazy-fixture-0.6.3, django-3.6.0
--Return--
[]
```

**Comment by jramosss** [comment]
Also, i tried removing the conftest.py, commenting all fixtures in that test module, and all tests within the class, only one thing worked.
If i do
```py
class TestIntegrationX:
    pass
```
it works, but if i do
```py
class TestIntegrationX:
    def test_example(self):
        assert True
```
It fails

**Comment by nicoddemus** [comment]
Can you try removing your plugins? You don't need even to run the full test suite, I think this is reproducible in your end by passing `--collect-only`?

**Comment by jramosss** [comment]
Well, it magically solved.
I uninstalled pytest-sugar, pytest-forked and pytest-xdist and worked.
Then i installed them all again and still works.
Thanks for your help!

**Comment by The-Compiler** [comment]
Assuming this was an issue in (perhaps an outdated version of) one of the plugins.

**Comment by eljrax** [comment]
In case it helps someone ending up here from a search - I had this same crash and it was fixed by updating pytest-sugar to 0.9.4 or later. I could trigger it by just running `pytest -k some_pattern`
